### ISSUE TITLE
Specifying cuda device through SupervisedRunner doesn't get picked up

### ISSUE THREAD

**Issue by the-lay** [catalyst-team/catalyst#478]
When you specify cuda device in SupervisedRunner (and SupervisedWandbRunner), it doesn't influence what GPUs will be used. Manually setting `CUDA_VISIBLE_DEVICES` works though.

**Example**
```python
import torch
from catalyst.dl.runner import SupervisedRunner, SupervisedWandbRunner
...
runner = SupervisedRunner(device=torch.device('cuda:1'))
runner.train(...)
```
Expected behavior: Only GPU with ID 1 should be used.
Actual behavior: Catalyst uses all available devices.

**Comment by TezRomacH** [comment]
Thanks, should be fixed in #482

**Comment by TezRomacH** [comment]
Fixed
